src/commonmark.rs

```diff
@@ -440,6 +440,68 @@ impl<'a, W: Write + 'a> Context<'a, W> {
         );
         write!(self.output.writer, "{}", osc(&command))
     }
+
+    /// Write an inline image for Terminology.
+    fn write_terminology_inline_image(&mut self, path: &Path) -> Result<()> {
+        fn get_lines(columns: u16, file: &str) -> u16 {
+            use std::process::Command;
+            use termion::terminal_size;
+
+            let lines = terminal_size().unwrap_or((0, 24)).1;
+
+            Command::new("identify")
+                .arg("-format")
+                .arg("%w %h")
+                .arg(file)
+                .output()
+                .and_then(|o| {
+                    fn get_v(x: Option<&str>) -> Result<u32> {
+                        x.and_then(|x| x.parse().ok())
+                            .ok_or(Into::<Error>::into(ErrorKind::InvalidData))
+                    }
+
+                    let o = String::from_utf8(o.stdout).or(Err(ErrorKind::InvalidData))?;
+                    let mut tokens = o.split(' ');
+                    let (w, h) = (get_v(tokens.next())? as f64, get_v(tokens.next())? as f64);
+
+                    // We divide by 2 because terminal cursor/font usually has a
+                    // 1:2 proportion
+                    Ok((h * (columns / 2) as f64 / w) as u16)
+                })
+                .unwrap_or(lines / 2)
+        }
+
+        fn is_http_url(path: &Path) -> bool {
+            path.to_str()
+                .map(|p| p.starts_with("http"))
+                .unwrap_or(false)
+        }
```

Comment:
Do we need this?  In the `if` below the behaviour doesn't change for HTTP URLs.

If we do need this, please extract into a `resources` helper module and use `Url` to implement it, ie, `Url::parse(…).map(|url| url.scheme == "https" || url.scheme == "http").unwrap_or(false)`.

Perhaps you can even generalize it into a `is_remote_reference` function, as in `Url::parse(…).map(|url| url.scheme != "file").unwrap_or(false)`.